### ISSUE TITLE
move ts-node to devDependencies

### ISSUE THREAD

**Issue by vorant94** [oclif/core#735]
**Describe the bug**
For some reason ts-node is included in regular dependencies, which results in annoying warning while using oclif with yarn:

```
➤ YN0000: ┌ Resolution step
➤ YN0002: │ @oclif/core@npm:2.9.3 doesn't provide @types/node (p3ea0d), requested by ts-node
➤ YN0002: │ @oclif/core@npm:2.9.3 doesn't provide typescript (pc0071), requested by ts-node
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 0s 280ms
```

**To Reproduce**
1. run `npx oclif generate mynewcli`
2. choose yarn as a package manager
3. re-run `yarn install` (optionally) 

**Expected behavior**
Since the actual code anyway is built to JS, there is no reason to mark `ts-node` as a regular dependency. It is only used in development and testing... maybe I'm missing something

**Screenshots**
![Screenshot 2023-07-14 at 10 23 21](https://github.com/oclif/core/assets/9719319/b3437577-e5b7-4aea-a58e-7a70344c98bf)

**Environment (please complete the following information):**
 - OS & version: macOS 13.4.1 (c) (22F770820d)
 - Shell/terminal & version zsh 5.9 (x86_64-apple-darwin22.0)

**Comment by cristiand391** [comment]
Hi @vorant94 

`ts-node` is also used to compile TS plugins at runtime when they are linked, see:
https://github.com/oclif/core/pull/650


plugin/command loading is done in `@oclif/core` so it needs `ts-node` as a dep.

**Comment by vorant94** [comment]
> Hi @vorant94 
> 
> 
> 
> `ts-node` is also used to compile TS plugins at runtime when they are linked, see:
> 
> https://github.com/oclif/core/pull/650
> 
> 
> 
> 
> 
> plugin/command loading is done in `@oclif/core` so it needs `ts-node` as a dep.

Is it acceptable to maybe make it a deepDep then?

**Comment by cristiand391** [comment]
> Is it acceptable to maybe make it a deepDep then?

You mean making ts-node a peerDependency?

**Comment by vorant94** [comment]
> > Is it acceptable to maybe make it a deepDep then?
> 
> You mean making ts-node a peerDependency?

yeah, sorry for typo

**Comment by SuperchupuDev** [comment]
any updates on this issue? it's a bit annoying to have ts-node always installed when it doesn't get used by any of my dependencies

**Comment by cristiand391** [comment]
hey, sorry missed the last response.

AFAIK, npm installs peerDependencies by default unless you do `npm i --omit peer`:
[npm v7 announcement](https://github.blog/2020-10-13-presenting-v7-0-0-of-the-npm-cli/#:~:text=Automatically%20installing%20peer%20dependencies%3A%20prior%20to%20npm%207%20developers%20needed%20to%20manage%20and%20install%20their%20own%20peer%20dependencies.%20The%20new%20peer%20dependency%20algorithm%20ensures%20that%20a%20validly%20matching%20peer%20dependency%20is%20found%20at%20or%20above%20the%20peer%2Ddependent%E2%80%99s%20location%20in%20the%20node_modules%20tree.)

Maybe we could remove `ts-node` from `@oclif/core` as dep and require plugins to have it as a devDep (which already do for `bin/dev` to work), but that would be a breaking change and might require some refactor to lazy-load ts-node code in core.

**Comment by mdonnalley** [comment]
Closed by #794
